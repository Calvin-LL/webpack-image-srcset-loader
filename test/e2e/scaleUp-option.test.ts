import WISLWebpackTestCompiler from "./helpers/WISLWebpackTestCompiler";

describe.each([4, 5] as const)('v%d "scaleUp" option', (webpackVersion) => {
  it("should not skip original size", async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["2912w"],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it("should skip greater size", async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["2913w", "1x"],
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });

  it("should work with true", async () => {
    const compiler = new WISLWebpackTestCompiler({ webpackVersion });
    const bundle = await compiler.compile({
      loaderOptions: {
        sizes: ["2913w"],
        scaleUp: true,
      },
    });

    expect(bundle.execute("main.js")).toMatchSnapshot("result");
  });
});
