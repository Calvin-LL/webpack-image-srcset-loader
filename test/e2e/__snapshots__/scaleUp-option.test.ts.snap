// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`v4 "scaleUp" option should not skip original size: result 1`] = `"bd2b9fdf768b5eb139702c4c3b4573de.jpeg 2912w"`;

exports[`v4 "scaleUp" option should skip greater size: result 1`] = `"bd2b9fdf768b5eb139702c4c3b4573de.jpeg 1x"`;

exports[`v4 "scaleUp" option should work with true: result 1`] = `"b9b3e9233f4da1756d2ea6b68b751583.jpeg 2913w"`;

exports[`v5 "scaleUp" option should not skip original size: result 1`] = `"bd2b9fdf768b5eb139702c4c3b4573de.jpeg 2912w"`;

exports[`v5 "scaleUp" option should skip greater size: result 1`] = `"bd2b9fdf768b5eb139702c4c3b4573de.jpeg 1x"`;

exports[`v5 "scaleUp" option should work with true: result 1`] = `"b9b3e9233f4da1756d2ea6b68b751583.jpeg 2913w"`;
